### PR TITLE
Integration testing 1 [P0]

### PR DESCRIPTION
**Pull request by oshhh**
Test 1 added:
Checks flow login->policy->discount->payment

1.  Start at login page
2. Find "Login with Google" button and click on it
3.  Find sidebar button and click on it. Select "Purchase Policy"
4.  Find address textboxes and fill them. Click on submit.
5.  Find radio buttons and policy names and select a policy
6.  Find "Smart Discounts" button and click on it
7.  Find "Add Devices" button and click on it
8.  Find the first a structure and select
9. TODO:  Find list of offers and confirm that only offers that can be availed are present
10.  Select an offer and click on "Go to Payment"
11. TODO:  Check if address, offers, structure and policies displayed are the ones chosen.
12.  Find "Pay" button and click on it
13.  Check if redirected to home page and Logout
14. Check if redirected to login screen (Failing currently: 3P app needs fix)

In order to run test: flutter drive --target=test_driver/app.dart

Working Video:
https://drive.google.com/file/d/1OoAr_eDOsiJE7hF0GAwxg2myiMRQGjku/view?usp=sharing

### CODE REVIEW

**Comment by oshhh** [comment]
@googlebot I consent.

File: home_insurance_app/lib/data/globals.dart [hunk 1]
```diff
@@ -3,14 +3,17 @@ library globals;
 import 'package:sdk/sdk.dart';
 import 'package:firebase_remote_config/firebase_remote_config.dart';
 
-Future<SDK> initialiseSDK() async {
+Future<SDK> initialiseSDK({test = 0}) async {
```
Comment (dreamplay) [inline home_insurance_app/lib/data/globals.dart:6]: can we skip test flag?
maybe use mockito for whatever it's required

File: home_insurance_app/test_driver/app_test.dart [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+FlutterDriver flutterDriver;
+
+// Clicks on LOGIN WITH GOOGLE button and checks if directed to homepage
+dynamic login() async {
+  SerializableFinder loginButton = find.text('LOG IN WITH GOOGLE');
+//      await Future.delayed(const Duration(seconds: 1));
+  await flutterDriver.runUnsynchronized(() async {
+    await flutterDriver.tap(loginButton);
```
Comment (dreamplay) [inline home_insurance_app/test_driver/app_test.dart:27]: This test will break as soon as we change text of button, can't we use some id for button and find using it here ?

Comment (dreamplay) [reply home_insurance_app/test_driver/app_test.dart:27]: If not an id, maybe use a substring case insensitive match

File: home_insurance_app/test_driver/app_test.dart [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  });
+
+  await flutterDriver.waitFor(find.byType("HomePage"));
+}
+
+// Opens app drawer and selects purchase policy tab. Checks if next page opens.
```
Comment (dreamplay) [inline home_insurance_app/test_driver/app_test.dart:33]: consider following above suggestion for all finds unless you're doing a find by some metadata that isnt shown to user and is not bound to change a lot

File: home_insurance_app/test_driver/app_test.dart [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+  SerializableFinder submitButton = find.text("Go to Payment");
+  await flutterDriver.tap(submitButton);
+
+  await flutterDriver.waitFor(find.byType("Payment"));
+//      await Future.delayed(const Duration(seconds: 1));
+}
+
+// Click on confirm payment on payments page. Check if redirected to home page
+dynamic confirmPayment() async {
```
Comment (dreamplay) [inline home_insurance_app/test_driver/app_test.dart:123]: redirection isn't enough, we'd like to check if entry in db was made for the purchase.

File: home_insurance_app/test_driver/app_test.dart [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+    //TODO:  Check if address, offers, structure and policies displayed are the ones chosen.
+
+    //  Find "Pay" button and click on it
+    test("Confirm Payment", confirmPayment);
+
+    // Check if redirected to home page and Logout
+    test("Logout", logout);
+  });
+}
```
Comment (dreamplay) [inline home_insurance_app/test_driver/app_test.dart:202]: nit: add new line
every file should end with a new line

File: sdk/lib/sdk.dart [hunk 6]
```diff
@@ -57,16 +58,35 @@ class SDK {
 class SDKBuilder {
   static SDK build(String clientId, String clientSecret, String enterpriseId,
       {Duration interactiveFlowTimeout = const Duration(minutes: 5),
-      Duration nonInteractiveFlowTimout = const Duration(seconds: 10)}) {
-    Login login = new Login(GoogleSignIn(),
+      Duration nonInteractiveFlowTimout = const Duration(seconds: 1),
+      testing = false}) {
```
Comment (dreamplay) [inline sdk/lib/sdk.dart:62]: try getting rid of this testing variable, maybe use a mockito

File: home_insurance_app/test_driver/app_test.dart (hunk 4 above)
Comment (oshhh) [reply home_insurance_app/test_driver/app_test.dart:123]: Right now entry to DB hasn't been integrated to master so its not there in the integration test.

File: sdk/lib/sdk.dart (hunk 6 above)
Comment (oshhh) [reply sdk/lib/sdk.dart:62]: Discussed offline. Adding a TODO for after demo change.

File: home_insurance_app/lib/data/globals.dart (hunk 1 above)
Comment (oshhh) [reply home_insurance_app/lib/data/globals.dart:6]: Discussed offline adding a TODO for after demo change.

**Comment by khushi2305** [comment]
@googlebot I consent.

**Comment by anjalisoni3655** [comment]
@googlebot I consent.

**Pull request merged**